src/types/privatekubectl.ts

```diff
@@ -0,0 +1,143 @@
+import * as os from 'os'
+import * as fs from 'fs'
+
+export class PrivateKubectl extends Kubectl {
+   public isPrivate(): boolean {
+      return true
+   }
+
+   protected async execute(args: string[], silent: boolean = false) {
+      args.unshift('kubectl')
+      let kubectlCmd = args.join(' ')
+      let addFileFlag = false
+      let eo = <ExecOptions>{silent}
+
+      if (this.containsFilenames(kubectlCmd)) {
+         core.debug('kubectl command contains filenames: ' + kubectlCmd)
+         // For private clusters, files will not be in the tmp directory
+         kubectlCmd = kubectlCmd.replace(/[\/][t][m][p][\/]/g, '')
+         core.debug(
+            'Filenames when invoking for private clusters: ' + kubectlCmd
+         )
+         addFileFlag = true
+      }
+
+      const privateClusterArgs = [
+         'aks',
+         'command',
+         'invoke',
+         '--resource-group',
+         this.resourceGroup,
+         '--name',
+         this.name,
+         '--command',
+         kubectlCmd
+      ]
+
+      if (addFileFlag) {
+         const filenames = this.extractFilesnames(kubectlCmd).split(' ')
+         const tempDirectory =
+            process.env['runner.tempDirectory'] || os.tmpdir() + '/manifests'
+         eo.cwd = tempDirectory
+         core.debug('ExecOptions current working directory: ' + eo.cwd)
+         privateClusterArgs.push(...['--file', '.'])
+
+         let filenamesArr = filenames[0].split(',')
+         for (let index = 0; index < filenamesArr.length; index++) {
+            var file = filenamesArr[index]
+
+            if (!file) {
+               continue
+            }
+            core.debug('Attempting to move file: ' + file)
+            this.moveFileToTempManifestDir(file)
+         }
+      }
+
+      core.debug(
+         `private cluster Kubectl run with invoke command: ${kubectlCmd}`
+      )
+      core.debug('EO as it goes into getExec ' + eo.cwd)
+      return await getExecOutput('az', privateClusterArgs, eo)
+   }
+
+   public extractFilesnames(strToParse: string) {
+      console.log('string to parse extractFiles: ' + strToParse)
+      var start = strToParse.indexOf('-filename')
+      var offset = 7
+
+      if (start == -1) {
+         start = strToParse.indexOf('-f')
+
+         if (start == -1) {
+            return ''
+         }
+         offset = 0
+      }
+
+      var temp = strToParse.substring(start + offset)
+      var end = temp.indexOf(' -')
+
+      //End could be case where the -f flag was last, or -f is followed by some additonal flag and it's arguments
+      return temp.substring(3, end == -1 ? temp.length : end).trim()
+   }
+
+   private containsFilenames(str: string) {
+      return str.includes('-f ') || str.includes('filename ')
+   }
+
+   private createTempManifestsDirectory() {
+      if (!fs.existsSync('/tmp/manifests')) {
+         try {
+            fs.mkdirSync('/tmp/manifests', {recursive: true})
+         } catch (e) {
+            core.debug(
```

Comment:
we should just throw an error if we can't make temp. Not a state we can recover from?